cstring.h

```diff
@@ -363,6 +363,40 @@ int64_t stringFindFirstCharArr(TString s, const char *pattern) {
     return -1; 
 }
 
+double stringToDouble(TString s) {
+    double result = 0.0;
+    int sign = 1;
+    int decimalIndex = -1;
+    if (s.size > 0 && s.data[0] == '-') {
+        sign = -1;
+        stringRemove(&s, 0, 1);
+    }
+    for (size_t i = 0; i < s.size; ++i) {
+        if (s.data[i] == '.') {
+            decimalIndex = i;
+            break;
+        }
+    }
+    for (size_t i = 0; i < s.size; ++i) {
+        if (s.data[i] == '.') {
+            break;
+        }
+        if (stringCharIsDigit(s.data[i])) {
+            result = result * 10.0 + (s.data[i] - '0');
+        }
```

Comment:
`Else` (when it's not a digit) set error like "Not a Number" using `setError()`

About error handling [here](https://github.com/Tnirpps/cstring/pull/5#discussion_r1605905205)